persist/fs/msgpack/decoder.go

```diff
@@ -91,7 +97,7 @@ func (dec *Decoder) DecodeIndexEntry() (schema.IndexEntry, error) {
 	if dec.err != nil {
 		return emptyIndexEntry, dec.err
 	}
-	numFieldsToSkip := dec.decodeRootObject(indexEntryVersion, indexEntryType)
+	_, numFieldsToSkip := dec.decodeRootObject(indexEntryVersion, indexEntryType)
```

Comment:
I know the original code isn't doing this, but I don't follow the rationale. Why don't you need to check `dec.err` after each `decodeRootObject()` call?

Reply:
again, not caused by your PR, but would it be better to just have all the helper fns return error in the signature? that way it's a lot more apparent what error handling needs to be done

Reply:
Talked to @xichen2020  about it, and it does actually work as intended. The err will get checked in the next methods that are called and will properly be handled there, even if its not super clear.

Since it works as intended, I'm gonna leave this as is and not do a big refactor here

Reply:
I also deleted my unnecessary err check to stay consistent with the rest of the methods